File: examples/python/send_single_sms.py
"""
MozeSMS API - Send Single SMS

Example showing how to send a single SMS message
"""

import requests
import config

def send_sms(phone: str, message: str, sender_id: str = None) -> dict:
    """
    Send a single SMS message
    
    Args:
        phone: Recipient phone number (258XXXXXXXXX format)
        message: Message content
        sender_id: Optional sender ID
        
    Returns:
        dict: Response data with success status
    """
    url = f"{config.BASE_URL}/v2/sms/send"
    
    headers = {
        'Content-Type': 'application/json',
        'X-API-Key': config.API_KEY,
        'X-API-Secret': config.API_SECRET
    }
    
    data = {
        'phone': phone,
        'message': message
    }
    
    if sender_id:
        data['sender_id'] = sender_id
    
    try:
        response = requests.post(
            url, 
            json=data, 
            headers=headers,
            timeout=(config.CONNECT_TIMEOUT, config.TIMEOUT)
        )
        
        return {
            'success': response.status_code == 200,
            'http_code': response.status_code,
            'data': response.json()
        }
        
    except requests.exceptions.RequestException as e:
        return {
            'success': False,
            'error': str(e)
        }

if __name__ == '__main__':
    # Example usage
    phone = '258840000000'  # Replace with recipient's phone number
    message = 'Olá! Esta é uma mensagem de teste do MozeSMS API.'
    sender_id = config.SENDER_ID
    
    print('Sending SMS...')
    print(f'Phone: {phone}')
    print(f'Message: {message}\n')
    
    result = send_sms(phone, message, sender_id)
    
    if result['success']:
        print('✅ SMS sent successfully!')
        print(f"Message ID: {result['data']['message_id']}")
        print(f"Status: {result['data']['status']}")
    else:
        print('❌ Failed to send SMS')
        error = result.get('error') or result.get('data', {}).get('message', 'Unknown error')
        print(f'Error: {error}')
    
    print('\nFull Response:')
    print(result)

File: examples/nodejs/send_single_sms.js
/**
 * MozeSMS API - Send Single SMS
 * 
 * Example showing how to send a single SMS message
 */

const axios = require('axios');
const config = require('./config');

/**
 * Send a single SMS message
 */
async function sendSMS(phone, message, senderId = null) {
    const url = `${config.baseUrl}/v2/sms/send`;
    
    const data = {
        phone,
        message
    };
    
    if (senderId) {
        data.sender_id = senderId;
    }
    
    try {
        const response = await axios.post(url, data, {
            headers: {
                'Content-Type': 'application/json',
                'X-API-Key': config.apiKey,
                'X-API-Secret': config.apiSecret
            },
            timeout: config.timeout
        });
        
        return {
            success: true,
            httpCode: response.status,
            data: response.data
        };
        
    } catch (error) {
        return {
            success: false,
            httpCode: error.response?.status,
            error: error.response?.data?.message || error.message,
            data: error.response?.data
        };
    }
}

// Example usage
async function main() {
    const phone = '258840000000'; // Replace with recipient's phone number
    const message = 'Olá! Esta é uma mensagem de teste do MozeSMS API.';
    const senderId = config.senderId;
    
    console.log('Sending SMS...');
    console.log(`Phone: ${phone}`);
    console.log(`Message: ${message}\n`);
    
    const result = await sendSMS(phone, message, senderId);
    
    if (result.success) {
        console.log('✅ SMS sent successfully!');
        console.log(`Message ID: ${result.data.message_id}`);
        console.log(`Status: ${result.data.status}`);
    } else {
        console.log('❌ Failed to send SMS');
        console.log(`Error: ${result.error}`);
    }
    
    console.log('\nFull Response:');
    console.log(result);
}

// Run if called directly
if (require.main === module) {
    main().catch(console.error);
}

module.exports = { sendSMS };
